modules/gapi/src/compiler/gcompiled.cpp

```diff
@@ -56,6 +56,11 @@ void cv::GCompiled::Priv::checkArgs(const cv::gimpl::GRuntimeArgs &args) const
                                            "for different metadata!"));
         // FIXME: Add details on what is actually wrong
     }
+    if (!validate_input_arg(args.inObjs))
+    {
+        util::throw_error(std::logic_error("This object was compiled "
+                                           "for incorrect input arguments!"));
```

Comment:
This text is wrong. The object has been compiled already, here you enter the execution.

Reply:
Removed